### DISCUSSION TITLE
Learning to Use the Terminal

### DISCUSSION THREAD

**Discussion by joehhn** [sn4k3/UVtools#665] [Q&A] [answered]
Hello! I recently discovered that UVTools has a terminal for accessing file information. What can I do with the terminal? Can I edit layer or file data, or is it all read-only? And more importantly, how can I see the available functions/commands? I would like to fetch UVTool's times for various aspects of the print (waiting, curing, etc.) instead of summing the layer times all myself. Thanks in advance!

**Comment by sn4k3** [comment, marked as answer]
You can access and edit everything inside the file format: `SlicerFile`
All public methods can be used, see them [here](https://github.com/sn4k3/UVtools/blob/master/UVtools.Core/FileFormats/FileFormat.cs)

eg: `SlicerFile[0].ExposureTime = 30f; // Modify exposure time for layer 0`

All others C# commands are also available

**Reply by joehhn** [reply]
Excellent, thanks!
